Fixes https://github.com/NuGet/Home/issues/420

alpaix opened the pull request:
Fixes https://github.com/NuGet/Home/issues/420.
- Updated extracted file LastWriteTime with a value in corresponding
  ZipArchiveEntry.
- Added unit-test to verify the fix.

yishaigalatzer commented on src/NuGet.Core/NuGet.Test.Utility/TestPackages.cs:
```diff
@@ -539,70 +543,70 @@ public static Stream GetTestPackageWithContentXmlFile()
         }
 
-        public static FileInfo GetPackageWithSHA512AtRoot(string path, string packageId, string packageVersion)
+        public static async Task<FileInfo> GetPackageWithSHA512AtRoot(string path, string packageId, string packageVersion)
```
there is not much point in making tests fully async, it forces all consumers to go async for no strong reason

alpaix replied:
Spotted the only consumer in a test that was async before the change. Besides GeneratePackageAsync method hides async stream writing so that entire call chain has to be async.

yishaigalatzer commented on src/NuGet.Core/NuGet.Test.Utility/TestPackages.cs:
```diff
@@ -518,9 +522,9 @@ public static Stream GetTestPackageWithContentXmlFile()
-                zip.AddEntry("lib/a.dll", new byte[] { 0 });
-                zip.AddEntry("[Content_Types].xml", new byte[] { 0 });
-                zip.AddEntry("content/[Content_Types].xml", new byte[] { 0 });
+                zip.AddEntry("lib/a.dll", ZeroContent);
```
nice cleanup - it's probably better to push it in a separate commit

alpaix replied:
Agree. Will do.

yishaigalatzer commented:
:shipit:

alpaix commented:
File NuGet.Client/src/NuGet.Core/NuGet.Common/ZipArchiveExtensions.cs doesn't seem to be used anywhere. So I went ahead and deleted it to avoid confusion and code duplication.

yishaigalatzer commented:
About the ziparchive, did you get a full build + tests passing without it? Common will not show references

alpaix commented:
@yishaigalatzer Full build as always! No shortcuts)))

alpaix committed:
Core package tests cleanup

+ Introduced test constant for zero byte entry content
+ Added async counterparts for AddEntry methods
+ Added AssertDirectoryExists and AssertFileExists for convenience

alpaix committed:
Fixes https://github.com/NuGet/Home/issues/420

Fixes https://github.com/NuGet/Home/issues/420
+ Updated extracted file LastWriteTime with a value in corresponding
ZipArchiveEntry.
+ Added unit-test to verify the fix.
+ Removed redundant file
NuGet.Client/src/NuGet.Core/NuGet.Common/ZipArchiveExtensions.cs

yishaigalatzer commented:
:+1:

alpaix committed:
Fixing test failure
